app/models/concerns/foreman_openscap/host_extensions.rb

```diff
@@ -45,9 +45,11 @@ def last_report_for_policy(policy)
 
     def reports_for_policy(policy, limit = nil)
       if limit
-        ForemanOpenscap::ArfReport.where(:asset_id => asset.id, :policy_id => policy.id).limit limit
+        ForemanOpenscap::ArfReport.joins(:policy_arf_report)
+          .merge(ForemanOpenscap::PolicyArfReport.of_policy policy.id).where(:host_id => asset.id).limit limit
```

Comment:
@shlomizadok: It is correct as it is, host_id in arf_report points to asset, same goes for the following cases